internal/ast/ast.go

```diff
@@ -0,0 +1,32 @@
+package ast
+
+type Node struct {
+	Position int `yaml:"pos"`
+}
+
+type Path []Step
+
+type Step struct {
+	Node `yaml:",inline"`
+
+	Index    *int    `yaml:"idx,omitempty"`
+	Identity *string `yaml:"id,omitempty"`
+}
+
+type Argument struct {
+	Node `yaml:",inline"`
+
+	Number *int    `yaml:"number,omitempty"`
+	String *string `yaml:"string,omitempty"`
+	Path   Path    `yaml:"path,omitempty"`
+}
+
+type Command struct {
+	Node `yaml:",inline"`
+
+	Path   Path       `yaml:"path,omitempty"`
+	Equal  []Argument `yaml:"equal,omitempty"`
```

Comment:
Same a the comment of the previous commit. We should be algined with the design doc. There it is called - "Filter".

Reply:
Done at ast PR https://github.com/nmstate/nmpolicy/pull/7 with `EqFilter`.